setup.py

```diff
@@ -6,11 +6,18 @@
 )
 
 extras_require = {
+    # Separate hypothesis extra to ensure same version across multiple extras
+    # that require it.
+    'hypothesis': [
```

Comment:
Yeah I don't love adding an extra just to be used later in the file. I think it's unintentionally committing us to a public API. (`eth-abi[hypothesis]`).

Maybe we're better off just separating `extras_require` into sequential definitions, like:
```py
extras_require = {}
extras_require['test'] = [
  "pytest...",
# etc
```

More ideas about executing that, below.